frontend/javascripts/oxalis/view/viewport_status_indicator.js

```diff
@@ -1,39 +1,45 @@
 // @flow
 
 import * as React from "react";
+import _ from "lodash";
 
 import { WarningOutlined } from "@ant-design/icons";
 
 import { Tooltip } from "antd";
 import { getUnrenderableLayersForCurrentZoom } from "oxalis/model/accessors/dataset_accessor";
-import { getCurrentResolution } from "oxalis/model/accessors/flycam_accessor";
 import { usePolledState } from "libs/react_helpers";
 
 const { useState } = React;
 
 export default function ViewportStatusIndicator() {
-  const [unrenderableLayerNames, setUnrenderableLayerNames] = useState([]);
+  const [unrenderableLayersWithInfo, setUnrenderableLayersWithInfo] = useState([]);
   const [renderMissingDataBlack, setRenderMissingDataBlack] = useState(true);
-  const [currentResolution, setCurrentResolution] = useState([1, 1, 1]);
 
   usePolledState(state => {
     const newMissingLayersNames = getUnrenderableLayersForCurrentZoom(state);
-    setUnrenderableLayerNames(
-      newMissingLayersNames.map(layer =>
-        layer.category === "segmentation" ? "Segmentation" : layer.name,
-      ),
+    const newAdjustedMissingLayersNames = newMissingLayersNames.map(
+      ([layer, smallerOrHigherInfo]) =>
+        layer.category === "segmentation"
+          ? ["Segmentation", smallerOrHigherInfo]
+          : [layer.name, smallerOrHigherInfo],
     );
+    if (!_.isEqual(unrenderableLayersWithInfo, newAdjustedMissingLayersNames)) {
+      setUnrenderableLayersWithInfo(newAdjustedMissingLayersNames);
+    }
 
     setRenderMissingDataBlack(state.datasetConfiguration.renderMissingDataBlack);
-
-    setCurrentResolution(getCurrentResolution(state));
   });
-
-  if (unrenderableLayerNames.length === 0) {
+  if (unrenderableLayersWithInfo.length === 0) {
     return null;
   }
-  const pluralS = unrenderableLayerNames.length > 1 ? "s" : "";
-  const dontVerb = unrenderableLayerNames.length > 1 ? "don't" : "doesn't";
+  const unrenderableLayerNamesWithSmallerZoomstep = unrenderableLayersWithInfo
+    .filter(([, { smaller }]) => smaller)
+    .map(([layerName]) => layerName);
+  const pluralSForSmallerZoomstep = unrenderableLayerNamesWithSmallerZoomstep.length > 1 ? "s" : "";
+  const unrenderableLayerNamesWithHigherZoomstep = unrenderableLayersWithInfo
+    .filter(([, { higher }]) => higher)
+    .map(([layerName]) => layerName);
+  const pluralSForHigherZoomstep = unrenderableLayerNamesWithHigherZoomstep.length > 1 ? "s" : "";
```

Comment:
I kinda dislike the naming of the variables here. The advantage is, that the names are pretty explainatory. Do you have suggestions for better names here @philippotto?

Reply:
see my other comment